spec/dynamoid/transaction_write/update_spec.rb

```diff
@@ -224,6 +224,134 @@
           end
         }.to output('saving updating updated saved ').to_stdout
       end
+
+      context 'adds' do
+        it 'to nil' do
+          obj1 = klass.create!(name: 'one')
+          described_class.execute do |txn|
+            txn.update! obj1 do |u|
+              u.add(record_count: 5)
+            end
+          end
+          obj1_found = klass.find(obj1.id)
+          expect(obj1_found).to eql(obj1)
+          expect(obj1_found.record_count).to eql(5)
+        end
+      end
+
+      it 'to a value' do
+        obj1 = klass.create!(name: 'one', record_count: 10)
+        described_class.execute do |txn|
+          txn.update! obj1 do |u|
+            u.set(name: 'oneone')
+            u.add(record_count: 5)
+          end
+        end
+        obj1_found = klass.find(obj1.id)
+        expect(obj1_found).to eql(obj1)
+        expect(obj1_found.name).to eql('oneone')
+        expect(obj1_found.record_count).to eql(15)
+      end
+
+      it 'an array to a set' do
+        obj1 = klass.create!(name: 'one', favorite_numbers: [1, 2, 3])
+        described_class.execute do |txn|
+          txn.update! obj1 do |u|
+            u.set(name: 'oneone')
+            u.add(favorite_numbers: [4]) # must be enumerable
+          end
+        end
+        obj1_found = klass.find(obj1.id)
+        expect(obj1_found).to eql(obj1)
+        expect(obj1_found.name).to eql('oneone')
+        expect(obj1_found.favorite_numbers).to eql(Set[1, 2, 3, 4])
+      end
+
+      it 'a set to a set' do
+        obj1 = klass.create!(name: 'one', favorite_numbers: [1, 2, 3])
+        described_class.execute do |txn|
+          txn.update! obj1 do |u|
+            u.set(name: 'oneone')
+            u.add(favorite_numbers: Set[3, 4]) # must be enumerable
+          end
+        end
+        obj1_found = klass.find(obj1.id)
+        expect(obj1_found).to eql(obj1)
+        expect(obj1_found.name).to eql('oneone')
+        expect(obj1_found.favorite_numbers).to eql(Set[1, 2, 3, 4])
+      end
+
+      it 'a set to a set of strings' do
+        obj1 = klass.create!(name: 'one', favorite_names: %w[adam ben charlie])
+        described_class.execute do |txn|
+          txn.update! obj1 do |u|
+            u.set(name: 'oneone')
+            u.add(favorite_names: Set['charlie', 'dan']) # must be enumerable
+          end
+        end
+        obj1_found = klass.find(obj1.id)
+        expect(obj1_found).to eql(obj1)
+        expect(obj1_found.name).to eql('oneone')
+        expect(obj1_found.favorite_names).to eql(Set.new(%w[adam ben charlie dan]))
+      end
+    end
```

Comment:
It makes sense to have a separate `context` for the set operation.

Reply:
I added more contexts within 'add':
```
updates
...
      adds
        a value
          to an existing value
          to nil which defaults to zero
        to a set
          a set of strings
          an array
          a set of numbers
```

Reply:
I meant to test `add` and `set` operations separately, in different contexts and different test cases. Now they are tested in the same cases.

Reply:
OK. I made a new context just for setting and also added 2 tests to ensure setting and adding can happen in the same block.
```
      sets and adds in a block
        to an existing value
        an array
      adds
        a value
          to nil which defaults to zero
          to an existing value
        to a set
          a set of numbers
          an array
          a set of strings
      sets in a block
        a string
```